pkg/skaffold/debug/cnb.go

```diff
@@ -97,58 +148,51 @@ func updateForCNBImage(container *v1.Container, ic imageConfiguration, transform
+		// strip and then restore the "--"
 		ic.arguments = ic.arguments[1:]
 		return ic, func(transformed []string) []string {
 			return append([]string{"--"}, transformed...)
 		}
 	}
 
-	processType := "web" // default buildpacks process type
-	// the launcher accepts the first argument as a process type
-	if len(ic.arguments) == 1 {
-		processType = ic.arguments[0]
-	} else if value := ic.env["CNB_PROCESS_TYPE"]; len(value) > 0 {
-		processType = value
-	}
-
-	for _, p := range m.Processes {
-		if p.Type == processType {
-			// Direct: p.Command is the command and p.Args are the arguments
-			if p.Direct {
-				// Detect and unwrap `/bin/sh -c ...`-style command lines; GCP Buildpacks turn Procfiles into `/bin/bash -c ...`
-				if len(p.Args) >= 2 && isShDashC(p.Command, p.Args[0]) {
-					p.Command = p.Args[1]
-					p.Args = p.Args[2:]
-					// and fall through to script type below
-				} else {
-					ic.arguments = append([]string{p.Command}, p.Args...)
-					return ic, func(transformed []string) []string {
-						return append([]string{"--"}, transformed...)
-					}
-				}
-			}
-			// Script type: split p.Command, pass it through the transformer, and then reassemble in the rewriter.
-			if args, err := shell.Split(p.Command); err == nil {
-				ic.arguments = args
+	if p, clArgs, found := findCNBProcess(ic, m); found {
+		// Direct: p.Command is the command and p.Args are the arguments
+		if p.Direct {
+			// Detect and unwrap `/bin/sh -c ...`-style command lines.
+			// For example, GCP Buildpacks turn Procfiles into `/bin/bash -c ...`
+			if len(p.Args) >= 2 && isShDashC(p.Command, p.Args[0]) {
+				p.Command = p.Args[1]
+				p.Args = p.Args[2:]
+				// and fall through to script type below
 			} else {
-				ic.arguments = []string{p.Command}
-			}
-			return ic, func(transformed []string) []string {
-				// reassemble back into a script with arguments
-				return append([]string{shJoin(transformed)}, p.Args...)
+				args := append([]string{p.Command}, p.Args...)
+				args = append(args, clArgs...)
+				ic.arguments = args
+				return ic, func(transformed []string) []string {
+					return append([]string{"--"}, transformed...)
+				}
 			}
 		}
+		// Script type: split p.Command, pass it through the transformer, and then reassemble in the rewriter.
+		if args, err := shell.Split(p.Command); err == nil {
+			ic.arguments = args
+		} else {
+			ic.arguments = []string{p.Command}
+		}
+		return ic, func(transformed []string) []string {
+			// reassemble back into a script with arguments
+			result := append([]string{shJoin(transformed)}, p.Args...)
+			result = append(result, clArgs...)
+			return result
+		}
 	}
 
 	if len(ic.arguments) == 0 {
-		// indicates an image mis-configuration as we should have resolved the the
-		// CNB_PROCESS_TYPE (if specified) or `web`.
-		logrus.Warnf("no CNB launch found for %s/%s", ic.artifact, processType)
+		logrus.Warnf("no CNB launch found for %s", ic.artifact)
```

Comment:
would it possible to debug a container if CNB launch is not found?

Reply:
No.  At this point we have an CNB launcher entrypoint (either `/cnb/lifecycle/launcher` or `/cnb/process/xxx`), but no corresponding process found.  So we don't have a command-line should be executed.

Basically we'll return, the debug transforms won't be able to figure anything out, and the container / pod will remain un-touched.  And the execution of the container will then fail as no process definition will be found.